### PR TITLE
Remove ___always_inline

### PR DESCRIPTION
**Pull request by mppf**
Historically, we introduced ___always_inline as a way around errors that
GASNet builds got because GASNet flags include -Winline. That is not
the best solution, since we want C compilers to be able to make their
own good choices about inlining, and since ___always_inline is not
standard C or C++. Here we remove ___always_inline and replace it
with just plain inline. In order to solve the problem with GASNet builds
that include -Winline, we find-and-replace -Winline with the empty
string in runtime/src/comm/gasnet/Makefile.share.

We will need to monitor this change for any significant performance impact.

Passes test/release/examples/ with CHPL_COMM=gasnet

### CODE REVIEW

**Comment by mppf** [comment]
Passed full local testing

**Comment by mppf** [comment]
@gbtitus, could you review this patch please?

File: runtime/include/chpl-wide-ptr-fns.h [hunk 1]
```diff
@@ -37,7 +37,7 @@
 // chpl_error needed for some of the packed wide pointer error cases.
 #include "error.h"
 
-#define WIDE_PTR_INLINE ___always_inline
+#define WIDE_PTR_INLINE inline
```
Comment (gbtitus) [inline runtime/include/chpl-wide-ptr-fns.h]: I don't see any benefit of this WIDE_PTR_INLINE macro, but I do see a bit of obfuscation. Would you mind getting rid of it and changing all the uses to plain "inline" while you're here?

**Comment by gbtitus** [comment]
Looks good, and good to merge. It would be nice if you could tackle WIDE_PTR_INLINE as mentioned in the line comment before merging; I don't need to see it again if you do this.

**Comment by mppf** [comment]
I've verified that GASNet and local testing still pass after the last update.

**Pull request merged**